sfml ninja compilation error on windows 10 32bit

I have a problem during **_sfml_** compilation. And compilation command **_.\vcpkg install sfml_** outputs:

```
-- Configuring x86-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:14 (message):
    Command failed: C:/Projekty/vcpkg/downloads/cmake-3.8.0-win32-x86/bin/cmake.exe;C:/Projekty/vcpkg/buildtrees/libflac/src/flac-1.3.2;-DLIBFLAC_ARCHITECTURE=x86;-DBUILD_SHARED_LIBS=ON;-DVCPKG_TARGET_TRIPLET=x86-windows;-DCMAKE_CXX_FLAGS= /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP;-DCMAKE_C_FLAGS= /DWIN32 /D_WINDOWS /W3 /utf-8 /MP;-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON;-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON;-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON;-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE;-DCMAKE_VERBOSE_MAKEFILE=ON;-DVCPKG_APPLOCAL_DEPS=OFF;-DCMAKE_TOOLCHAIN_FILE=C:/Projekty/vcpkg/scripts/buildsystems/vcpkg.cmake;-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON;-DCMAKE_CXX_FLAGS_RELEASE=/MD /O2 /Oi /Gy /DNDEBUG /Zi;-DCMAKE_C_FLAGS_RELEASE=/MD /O2 /Oi /Gy /DNDEBUG /Zi;-DCMAKE_SHARED_LINKER_FLAGS_RELEASE=/DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF;-DCMAKE_EXE_LINKER_FLAGS_RELEASE=/DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF;-G;Ninja;-DCMAKE_BUILD_TYPE=Release;-DCMAKE_INSTALL_PREFIX=C:/Projekty/vcpkg/packages/libflac_x86-windows
    Working Directory: C:/Projekty/vcpkg/buildtrees/libflac/x86-windows-rel
    See logs for more information:
      C:\Projekty\vcpkg\buildtrees\libflac\config-x86-windows-rel-out.log
      C:\Projekty\vcpkg\buildtrees\libflac\config-x86-windows-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:100 (vcpkg_execute_required_process)
  ports/libflac/portfile.cmake:23 (vcpkg_configure_cmake)
  scripts/ports.cmake:73 (include)


Error: Building package libflac:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libflac:x86-windows
  Vcpkg version: 0.0.80-75cdbd398b18b413d55cf76a7286b3681ea0806f

Additionally, attach any relevant sections from the log files above.
```

An information from file config-x86-windows-rel-err.log is in Polish language and the translation is (_**This version does not correspond to the running Windows system. Check informations about the system on this PC and contact to software distributor**_):
```
CMake Error at CMakeLists.txt:4 (project):
  Running

   'C:/Projekty/vcpkg/downloads/tools/ninja/ninja-1.7.2/ninja.exe' '--version'

  failed with:

   Wersja %1 nie jest zgodna z wersj─ů uruchomionego systemu Windows. Sprawd┼║ informacje dotycz─ůce systemu operacyjnego na tym komputerze, a nast─Öpnie skontaktuj si─Ö z wydawc─ů oprogramowania

CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage

```
The problem occurs on Windows 10 32 bit and in my oppinion it is a problem with **_ninja executable binary_** which is in 64 bit version only.

Thanks for quick answer. I have made as you wrote. It seems that the problem is fix now. However ther is compilation error with libflac. Error message blow:
```

 Trwa tworzenie biblioteki C:/Projekty/vcpkg/buildtrees/libflac/x86-windows-rel/Release/flac.lib i obiektu C:/Projekty/vcpkg/buildtrees/libflac/x86-windows-rel/Release/flac.exp
     5>cpu.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__cpu_have_cpuid_asm_ia32 przywołany w funkcji _ia32_cpu_info [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_decoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_restore_signal_asm_ia32 przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_decoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_restore_signal_asm_ia32_mmx przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_decoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_restore_signal_wide_asm_ia32 przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__fixed_compute_best_predictor_asm_ia32_mmx_cmov przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_autocorrelation_asm_ia32 przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_autocorrelation_asm_ia32_sse_lag_4_old przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_autocorrelation_asm_ia32_sse_lag_8_old przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_autocorrelation_asm_ia32_sse_lag_12_old przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_autocorrelation_asm_ia32_sse_lag_16_old przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_residual_from_qlp_coefficients_asm_ia32 przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_residual_from_qlp_coefficients_asm_ia32_mmx przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_residual_from_qlp_coefficients_wide_asm_ia32 przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\Release\flac.dll : fatal error LNK1120: liczba nierozpoznanych elementów zewnętrznych: 13 [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
     5>Kompilowanie projektu "C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj" wykonane (domylne elementy docelowe) - NIEPOWODZENIE.
     3>Kompilowanie projektu "C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\ALL_BUILD.vcxproj" wykonane (domylne elementy docelowe) - NIEPOWODZENIE.
     1>Kompilowanie projektu "C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\install.vcxproj" wykonane (domylne elementy docelowe) - NIEPOWODZENIE.

Kompilacja NIE POWIODťA SI¨.

       "C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\install.vcxproj" (domylny element docelowy) (1)->
       "C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\ALL_BUILD.vcxproj" (domylny element docelowy) (3)->
       "C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj" (domylny element docelowy) (5)->
       (element docelowy ClCompile) -> 
         c:\projekty\vcpkg\buildtrees\libflac\src\flac-1.3.2\src\libflac\lpc_intrin_avx2.c(65): warning C4752: znaleziono rozszerzenie Intel(R) Advanced Vector Extensions; rozwaľ uľycie opcji /arch:AVX [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         c:\projekty\vcpkg\buildtrees\libflac\src\flac-1.3.2\src\libflac\lpc_intrin_avx2.c(768): warning C4752: znaleziono rozszerzenie Intel(R) Advanced Vector Extensions; rozwaľ uľycie opcji /arch:AVX [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         c:\projekty\vcpkg\buildtrees\libflac\src\flac-1.3.2\src\libflac\lpc_intrin_avx2.c(420): warning C4752: znaleziono rozszerzenie Intel(R) Advanced Vector Extensions; rozwaľ uľycie opcji /arch:AVX [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         C:\Projekty\vcpkg\buildtrees\libflac\src\flac-1.3.2\src\libFLAC\metadata_iterators.c(579): warning C4244: "return": konwersja z "const __int64" do "off_t", moľliwa utrata danych [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         C:\Projekty\vcpkg\buildtrees\libflac\src\flac-1.3.2\src\libFLAC\metadata_object.c(1127): warning C4244: "=": konwersja z "FLAC__uint64" do "unsigned int", moľliwa utrata danych [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         C:\Projekty\vcpkg\buildtrees\libflac\src\flac-1.3.2\src\libFLAC\metadata_iterators.c(1117): warning C4244: "+=": konwersja z "const __int64" do "unsigned int", moľliwa utrata danych [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         C:\Projekty\vcpkg\buildtrees\libflac\src\flac-1.3.2\src\libFLAC\metadata_iterators.c(1129): warning C4244: "=": konwersja z "__int64" do "unsigned int", moľliwa utrata danych [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         C:\Projekty\vcpkg\buildtrees\libflac\src\flac-1.3.2\src\libFLAC\metadata_iterators.c(1152): warning C4244: "-=": konwersja z "const __int64" do "unsigned int", moľliwa utrata danych [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         C:\Projekty\vcpkg\buildtrees\libflac\src\flac-1.3.2\src\libFLAC\metadata_iterators.c(1647): warning C4244: "=": konwersja z "__int64" do "unsigned int", moľliwa utrata danych [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         C:\Projekty\vcpkg\buildtrees\libflac\src\flac-1.3.2\src\libFLAC\metadata_iterators.c(1652): warning C4244: "=": konwersja z "__int64" do "unsigned int", moľliwa utrata danych [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         C:\Projekty\vcpkg\buildtrees\libflac\src\flac-1.3.2\src\libFLAC\metadata_iterators.c(1666): warning C4244: "=": konwersja z "const __int64" do "unsigned int", moľliwa utrata danych [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]


       "C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\install.vcxproj" (domylny element docelowy) (1)->
       "C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\ALL_BUILD.vcxproj" (domylny element docelowy) (3)->
       "C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj" (domylny element docelowy) (5)->
       (element docelowy Link) -> 
         cpu.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__cpu_have_cpuid_asm_ia32 przywołany w funkcji _ia32_cpu_info [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_decoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_restore_signal_asm_ia32 przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_decoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_restore_signal_asm_ia32_mmx przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_decoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_restore_signal_wide_asm_ia32 przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__fixed_compute_best_predictor_asm_ia32_mmx_cmov przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_autocorrelation_asm_ia32 przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_autocorrelation_asm_ia32_sse_lag_4_old przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_autocorrelation_asm_ia32_sse_lag_8_old przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_autocorrelation_asm_ia32_sse_lag_12_old przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_autocorrelation_asm_ia32_sse_lag_16_old przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_residual_from_qlp_coefficients_asm_ia32 przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_residual_from_qlp_coefficients_asm_ia32_mmx przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         stream_encoder.obj : error LNK2019: nierozpoznany zewnętrzny symbol _FLAC__lpc_compute_residual_from_qlp_coefficients_wide_asm_ia32 przywołany w funkcji _init_stream_internal_ [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
         C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\Release\flac.dll : fatal error LNK1120: liczba nierozpoznanych elementów zewnętrznych: 13 [C:\Projekty\vcpkg\buildtrees\libflac\x86-windows-rel\libFLAC.vcxproj]
```
I hope you will able to decode above code with error messages. I have Visual Studio 2017. It is probably **_libflac linking error_**. Compilation command was: **_.\vcpkg install sfml_**

--- a/ports/libflac/CMakeLists.txt
+++ b/ports/libflac/CMakeLists.txt
@@ -25,12 +25,13 @@ endif()
 if(LIBFLAC_ARCHITECTURE MATCHES x86)
     add_definitions(-DFLAC__CPU_IA32)
     add_definitions(-DFLAC__HAS_NASM)
-    enable_language(ASM_NASM)
-    list(APPEND LIBFLAC_SOURCES
-        src/libFLAC/ia32/cpu_asm.nasm
-        src/libFLAC/ia32/fixed_asm.nasm
-        src/libFLAC/ia32/lpc_asm.nasm)
-    set(CMAKE_ASM_NASM_FLAGS "-i\"${PROJECT_SOURCE_DIR}/src/libFLAC/ia32/\" -f win32 -d OBJ_FORMAT_win32")
+    file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/nasm)
+    foreach(ASM_SOURCE cpu_asm fixed_asm lpc_asm)
+        execute_process(COMMAND nasm "-i${PROJECT_SOURCE_DIR}/src/libFLAC/ia32/" -f win32 -d OBJ_FORMAT_win32 -f win32 
+            -o "${CMAKE_BINARY_DIR}/nasm/${ASM_SOURCE}.obj" "${PROJECT_SOURCE_DIR}/src/libFLAC/ia32/${ASM_SOURCE}.nasm"
+            WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+        list(APPEND LIBFLAC_SOURCES ${CMAKE_BINARY_DIR}/nasm/${ASM_SOURCE}.obj)
+    endforeach()
 elseif(LIBFLAC_ARCHITECTURE MATCHES x64)
     add_definitions(-DFLAC__CPU_X86_64)
     add_definitions(-DENABLE_64_BIT_WORDS)

--- a/ports/libflac/CONTROL
+++ b/ports/libflac/CONTROL
@@ -1,4 +1,4 @@
 Source: libflac
-Version: 1.3.2-1
+Version: 1.3.2-2
 Description: Library for manipulating FLAC files
 Build-Depends: libogg

